packages/caravan-fees/src/cpfp.ts

```diff
@@ -0,0 +1,301 @@
+    targetFeeRate: Number(txAnalyzer.cpfpFeeRate), // Use CPFP-specific fee rate from analyzer
+    dustThreshold,
+    network,
+    scriptType,
+    requiredSigners,
+    totalSigners,
+  });
+
+  // Step 4: Add the spendable output from the parent transaction as an input
+  const parentOutput = txAnalyzer.outputs[spendableOutputIndex];
+  if (!parentOutput) {
+    throw new Error(
+      `Spendable output at index ${spendableOutputIndex} not found in the parent transaction.`,
+    );
+  }
+
+  const parentOutputUTXO = availableInputs.find(
+    (utxo) => utxo.txid === analysis.txid && utxo.vout === spendableOutputIndex,
+  );
+
+  if (!parentOutputUTXO) {
+    throw new Error(
+      `UTXO for the spendable output (${analysis.txid}:${spendableOutputIndex}) not found in availableInputs.`,
+    );
+  }
+
+  childTxTemplate.addInput(BtcTxInputTemplate.fromUTXO(parentOutputUTXO));
+
+  // Step 5: Add a change output (at least 1 output required for a valid transaction)
+  childTxTemplate.addOutput(
+    new BtcTxOutputTemplate({
+      address: changeAddress,
+      amountSats: "0", // Initial amount, will be adjusted later
+    }),
+  );
+
+  /**
+   * Step 6: Add additional inputs to cover the CPFP fee requirements
+   *
+   * This step implements a simplified version of the "Child-pays-for-parent" (CPFP)
+   * transaction selection logic, inspired by Bitcoin Core's implementation
+   * (https://github.com/bitcoin/bitcoin/pull/7600).
+   *
+   * In Bitcoin Core, transactions are sorted in the mempool using a modified
+   * feerate that considers ancestors. During block creation, transactions are
+   * considered in this order. As transactions are selected, a separate map tracks
+   * the new feerate-with-ancestors for in-mempool descendants.
+   *
+   * Our simplified approach:
+   * 1. We start with the parent transaction's fee and vsize.
+   * 2. We iteratively add inputs to the child transaction.
+   * 3. After each input addition, we recalculate the combined fee rate of the
+   *    parent and child transactions.
+   * 4. We continue adding inputs until the combined fee rate meets or exceeds
+   *    the target fee rate.
+   *
+   * This method ensures that the child transaction provides enough fee to incentivize
+   * miners to include both the parent and child transactions in a block, effectively
+   * "bumping" the fee of the parent transaction.
+   *
+   * Note: While this approach is simpler than Bitcoin Core's full implementation,
+   * it achieves the core goal of CPFP: ensuring the combined package (parent + child)
+   * is attractive for miners to include in a block.
+   */
+
+  const parentFee = new BigNumber(analysis.fee);
+  const parentVsize = analysis.vsize;
+
+  for (const utxo of availableInputs) {
+    if (
+      isCPFPFeeSatisfied(
+        parentFee,
+        parentVsize,
+        new BigNumber(childTxTemplate.currentFee),
+        childTxTemplate.estimatedVsize,
+        parseFloat(txAnalyzer.cpfpFeeRate),
+      )
```

Comment:
Might be cleaner to let the isCPFPFeeSatisfied do its own conversion, then you can just do:

```suggestion
      isCPFPFeeSatisfied(
        parentFee,
        parentVsize,
        childTxTemplate.currentFee,
        childTxTemplate.estimatedVsize,
        txAnalyzer.cpfpFeeRate,
      )
```

I'd also suggest using an options object for the argument to reduce the complexity of the func sig.

Reply:
Actually, you could simplify this even more... what if the function just took an analyzer (which has all the parent info and the cpfpFeeRate) and the childTxTemplate? 
```suggestion
      isCPFPFeeSatisfied(txAnalyzer, childTxTemplate)
```

Reply:
changes [done](https://github.com/caravan-bitcoin/caravan/pull/114/commits/ed9f6c70ed681c17d1ed07b444329b98811948db#diff-852fd83c241ed78594dc29389ec04d1425d784e665c706434a8de046f7daa85dR250)